libraries/Microsoft.Bot.Builder/BotFrameworkAdapter.cs

```diff
@@ -932,5 +939,25 @@ private async Task<MicrosoftAppCredentials> GetAppCredentialsAsync(string appId,
             _appCredentialMap[appId] = appCredentials;
             return appCredentials;
         }
+
+        /// <summary>
+        /// Middleware to assign tenantId from channelData to Conversation.TenantId.
+        /// </summary>
+        internal class TenantIdWorkaroundForTeamsMiddleware : IMiddleware
+        {
+            public async Task OnTurnAsync(ITurnContext turnContext, NextDelegate next, CancellationToken cancellationToken = default(CancellationToken))
+            {
+                if (turnContext.Activity.ChannelId == Channels.Msteams && turnContext.Activity.Conversation != null && string.IsNullOrEmpty(turnContext.Activity.Conversation.TenantId))
+                {
```

Comment:
I would update the "==" compare against Channels.Msteams to be culture invariant.

Reply:
Addressed